src/praisonai-agents/tests/unit/context/test_context_compression.py

```
@@ -0,0 +1,172 @@
+        try:
+            if test_func():
+                passed += 1
+        except Exception as e:
+            print(f"✗ {test_func.__name__} failed with exception: {e}")
+    
+    print("============================================================")
+    print(f"Tests passed: {passed}/{total}")
+    
+    if passed == total:
+        print("🎉 All tests passed!")
+        sys.exit(0)
```

Ruff (0.15.15)

[error] Undefined name `sys`

(F821)